src/main/java/edu/wisc/my/restproxy/service/RestProxyServiceImpl.java

```diff
@@ -84,6 +98,19 @@ public Object proxyRequest(final String resourceKey, final HttpServletRequest re
       .setUri(uri.toString())
       .setUsername(username);
 
+    RequestBody requestBody = null;
+    try {
+      InputStream inputStream = request.getInputStream();
+      final String contentType = request.getHeader("Content-Type");
+      if(inputStream != null && "application/json".equals(contentType)) {
+        Object body = objectMapper.readValue(request.getInputStream(), Object.class);
```

Comment:
I've been thinking about this specific line - this is a super expensive price to pay for proxying. In the act of proxying we are deserializing the JSON, then serializing it back out again (hiding inside `RestTemplate#exchange`). I'm going to see if there isn't a way I can get `HttpServletRequest#getInputStream` into the HttpEntity's body field without paying the cost of passing it through Jackson.